miner/worker.go

```diff
@@ -87,11 +87,13 @@ var (
 // environment is the worker's current environment and holds all
 // information of the sealing block generation.
 type environment struct {
-	signer   types.Signer
-	state    *state.StateDB // apply state changes here
-	tcount   int            // tx count in cycle
-	gasPool  *core.GasPool  // available gas used to pack transactions
-	coinbase common.Address
+	signer        types.Signer
+	state         *state.StateDB     // apply state changes here
+	tcount        int                // tx count in cycle
+	gasPool       *core.GasPool      // available gas used to pack transactions
+	multiGasPool  *core.MultiGasPool // available per-fee-currency gas used to pack transactions
+	exchangeRates common.ExchangeRates
```

Comment:
Looks like exchange rates are not used anywhere directly, only to obtain the whitelist. Why not store the whitelist then? It can safely be an empty slice which should help with the FIXME below.

Reply:
Yes sorry, the FIXME was an accidental checkin that was not an issue after all (the empty map would have worked in the same way an empty slice does).
However I still changed it to store the whitelist directly.

Reply:
Looks good to me!